manual_test_functions.py

```diff
@@ -25,6 +27,14 @@ async def main() -> None:
     group = await s.get_group(random_group_id)
     print(f"{_group_summary(group)}")
 
+    print("\nGetting a nonexistent group by id to check exception handling...")
+    try:
+        await s.get_group(DUMMY_ID)
+    except IndexError as error:
+        print(f"Exception raised: {error!r}")
```

Comment:
Can we use `pytest` here to test for the exception being raised?

See [example](https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions)

Reply:
Yes you are right here, I will factor out to a separate test file though. It's easily mocked and doesn't belong with the 'just summarise the data return' purpose of this script.